homeassistant/components/rejseplanen/sensor.py

```diff
@@ -6,252 +6,254 @@
-CONF_ROUTE = "route"
-CONF_DIRECTION = "direction"
-CONF_DEPARTURE_TYPE = "departure_type"
-
-DEFAULT_NAME = "Next departure"
-
-
-SCAN_INTERVAL = timedelta(minutes=1)
-
-BUS_TYPES = ["BUS", "EXB", "TB"]
-TRAIN_TYPES = ["LET", "S", "REG", "IC", "LYN", "TOG"]
-METRO_TYPES = ["M"]
-
-PLATFORM_SCHEMA = SENSOR_PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_STOP_ID): cv.string,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_ROUTE, default=[]): vol.All(cv.ensure_list, [cv.string]),
-        vol.Optional(CONF_DIRECTION, default=[]): vol.All(cv.ensure_list, [cv.string]),
-        vol.Optional(CONF_DEPARTURE_TYPE, default=[]): vol.All(
-            cv.ensure_list, [vol.In([*BUS_TYPES, *TRAIN_TYPES, *METRO_TYPES])]
-        ),
-    }
-)
-
-
-def due_in_minutes(timestamp):
-    """Get the time in minutes from a timestamp.
-
-    The timestamp should be in the format day.month.year hour:minute
-    """
-    diff = datetime.strptime(timestamp, "%d.%m.%y %H:%M") - dt_util.now().replace(
-        tzinfo=None
-    )
-
-    return int(diff.total_seconds() // 60)
-
 
-def setup_platform(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
-    add_devices: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+    config_entry: RejseplanenConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
 ) -> None:
     """Set up the Rejseplanen transport sensor."""
-    name = config[CONF_NAME]
-    stop_id = config[CONF_STOP_ID]
-    route = config.get(CONF_ROUTE)
-    direction = config[CONF_DIRECTION]
-    departure_type = config[CONF_DEPARTURE_TYPE]
-
-    data = PublicTransportData(stop_id, route, direction, departure_type)
-    add_devices(
-        [RejseplanenTransportSensor(data, stop_id, route, direction, name)], True
+    coordinator: RejseplanenDataUpdateCoordinator = config_entry.runtime_data
+
+    for subentry_id, subentry in config_entry.subentries.items():
+        _LOGGER.debug("Subentry %s with data: %s", subentry_id, subentry.data)
+
+        config = subentry.data
+        name = config.get(CONF_NAME, DEFAULT_STOP_NAME)
+        stop_id = int(config[CONF_STOP_ID])
+        route = config.get(CONF_ROUTE, [])
+        direction = config.get(CONF_DIRECTION, [])
+        departure_type = config.get(CONF_DEPARTURE_TYPE, [])
+
+        async_add_entities(
+            [
+                RejseplanenTransportSensor(
+                    coordinator=coordinator,
+                    stop_id=stop_id,
+                    entry_id=config_entry.entry_id,
+                    name=name,
+                    route=route,
+                    direction=direction,
+                    departure_type=departure_type,
+                )
+            ],
+            config_subentry_id=subentry_id,
+        )
+
+    await (
+        coordinator.async_refresh()
+    )  # otherwise it will take approx 5 minutes to get data from API
```

Comment:
Instead, run `await coordinator.async_config_entry_first_refresh()`in `__init__.py` before passing onto the platforms. This makes sure that the coordinator has a state